src/category_theory/limits/limits.lean

```diff
@@ -933,6 +1024,22 @@ def has_colimit.of_cocones_iso {J K : Type v} [small_category J] [small_category
   (h : F.cocones ≅ G.cocones) [has_colimit F] : has_colimit G :=
 ⟨_, is_colimit.of_nat_iso ((is_colimit.nat_iso (colimit.is_colimit F)) ≪≫ h)⟩
 
+/--
+The chosen colimits of `F : J ⥤ C` and `G : J ⥤ C` are isomorphic,
+if the functors are naturally isomorphic.
+-/
+def has_colimit.ext_of_nat_iso {F G : J ⥤ C} [has_colimit F] [has_colimit G] (w : F ≅ G) :
```

Comment:
To me, this "feels" more like a `congr`  rather than an `ext`.

Reply:
Hmm... it occurs to me I don't understand what the line between these is at all!

Reply:
FWIW, here's how I think of these:

`ext` is for things that are in some way made out of smaller things. It reduces equality of composite objects to equality of each component. Examples: `funext`, `is_limit.hom_ext` (as illustrated by `prod.hom_ext`), `ext` attribute for structures.

`congr`, on the other hand, to me is the idea that applying an operation to (in some sense) equal things should give results that are (in some sense) equal. Examples: `congr_arg`, the `congr` tactic, `fintype.card_congr`.

I don't know whether these definitions are close to what these terms really are supposed to mean, but if they are, then I think the definition should be a `congr`: If `F` and `G` are naturally isomorphic, then taking colimits yields isomorphic objects.

Reply:
Okay!

How about I split the difference and call this `has_colimit.iso_of_nat_iso`?